### PR TITLE
[kdalgorithms] Add new port

### PR DESCRIPTION
**Pull request by hmoffatt**
This is a header-only library with some stl algorithm wrappers.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [X] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [X] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [X] The versioning scheme in `vcpkg.json` matches what upstream says.
- [X] The license declaration in `vcpkg.json` matches what upstream says.
- [X] The installed as the "copyright" file matches what upstream says.
- [X] The source code of the component installed comes from an authoritative source.
- [X] The generated "usage text" is accurate. See docs/examples/adding-an-explicit-usage.md for context.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is in the new port's versions file.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/kdalgorithms/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+vcpkg_from_github(
```
Comment (LilyWangLL) [inline ports/kdalgorithms/portfile.cmake:2]:
```suggestion
# header-only library
vcpkg_from_github(
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Review by LilyWangLL — DISMISSED** [review]

File: ports/kdalgorithms/usage [hunk 2]
```diff
@@ -0,0 +1,4 @@
+kdalgorithms is header-only and can be used from CMake via:
+
+    find_path(KDALGORITHMS_INCLUDE_DIRS "kdalgorithms/kdalgorithms.h")
+    target_include_directories(main PRIVATE ${KDALGORITHMS_INCLUDE_DIRS}/kdalgorithms)
```
Comment (dg0yt) [inline ports/kdalgorithms/usage:4]: This is counter-intuitive. In general practice, `<Pkg>_INCLUDE_DIRS` contains the full path (or even multiple paths) to be passed to `target_link_libraries`.
So if the expected usage is `include <kdalgorithms.h>`, then literally use this filename with `find_path` and provide the expected subdir via `PATH_SUFFIXES kdalgorithms`.

Comment (hmoffatt) [reply ports/kdalgorithms/usage]: Thank you, fixed.

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**